### PR TITLE
feat(argo-rollouts): Support revisionHistoryLimit

### PR DESCRIPTION
**Pull request by yu-croco**
Resolves https://github.com/argoproj/argo-helm/issues/2711

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by mkilchhofer** [comment]
You are so fast @yu-croco ⚡️ my plan was to file a PR today🤣

In argo-cd chart we only support global setting for all components:
https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-repo-server/deployment.yaml#L22

Shall we implement it in rollouts different? Are there use cases for different settings from component to component?

**Comment by yu-croco** [comment]
Ah right, let's follow Argo CD's, I will fix soon. 🚀 

~I hope nobody realize that I am totally available any time.~ 🙈

File: charts/argo-rollouts/values.yaml [hunk 1]
```diff
@@ -43,6 +43,8 @@ global:
   deploymentAnnotations: {}
   # -- Labels for all deployed Deployments
   deploymentLabels: {}
+  # -- Number of old deployment ReplicaSets to retain. The rest will be garbage collected.
+  revisionHistoryLimit: 10
```
Comment (yu-croco) [inline charts/argo-rollouts/values.yaml:47]: Now default is `10` 📝 
Ref: https://kubernetes.io/docs/reference/using-api/deprecation-guide/

image

**Review by jmeridth — APPROVED** [review]

**Review by tico24 — APPROVED** [review]

**Review by mkilchhofer — APPROVED** [review]
LGTM, 🚢 it :)

**Pull request merged**